h/services/bulk_annotation.py

```diff
@@ -170,18 +170,20 @@ def _field_names_to_columns(cls, field_names):
 
     @classmethod
     def _audience_groups_subquery(cls, authority, audience):
-        return sa.select(Group.id).join(
-            cls._AUDIENCE,
-            Group.members.any(
-                # pylint: disable=protected-access
-                User._username.in_(
+        return (
+            sa.select(Group.id)
+            .distinct()
+            .join(GroupMembership, GroupMembership.group_id == Group.id)
+            .join(cls._AUDIENCE, GroupMembership.user_id == cls._AUDIENCE.id)
+            .where(
+                cls._AUDIENCE._username.in_(  # pylint:disable=protected-access
                     [
                         username.lower().replace(".", "")
                         for username in audience["username"]
                     ]
                 ),
-                authority=authority,
-            ),
+                cls._AUDIENCE.authority == authority,
+            )
```

Comment:
This is the main fix: explicitly doing the two-step `JOIN` through `GroupMembership` (the `user_group` table) and then doing the `WHERE` seperately gets SQLAlchemy to generate sensible SQL